src/Handler/Command.hs

```diff
@@ -56,6 +59,9 @@ patchCommandR token = do
 getCommandR :: Token -> Handler TypedContent
 getCommandR token = do
     command <- unsafeRunStorage $ get404 token
+    mcontent <- if commandArchived command
+        then Just <$> getOutput token sinkLbs
```

Comment:
`pure = Just` so you could use `pure <$> getOutput token sinkLbs`, though I don't find it particularly readable.

```hs
λ> :t (pure <$>)
(pure <$>) :: (Functor f, Applicative f1) => f a -> f (f1 a)
-- f  :: Handler
-- f1 :: Maybe
-- a  :: ByteString
```

`pure = return`, so `(pure =<<)` is actually just a form of `id`:

```hs
λ> :t (pure =<<)
(pure =<<) :: Monad m => m b -> m b
```

Which makes sense, you're pulling the value out with `=<<` then putting it back with `pure`. So that would be a type error since `Handler ByteString != Handler (Maybe ByteString)`

That said, Joe's sum-type idea will make this go away anyway.